src/main/java/net/sf/jabref/logic/importer/fetcher/MrDLibFetcher.java

```diff
@@ -0,0 +1,135 @@
+        String response = makeServerRequest(entry.getLatexFreeField("title").get());
+        MrDLibImporter importer = new MrDLibImporter();
+        ParserResult parserResult = new ParserResult();
+        try {
+            if (importer.isRecognizedFormat(new BufferedReader(new StringReader(response)))) {
+                parserResult = importer.importDatabase(new BufferedReader(new StringReader(response)));
+            } else {
+                // For displaying An ErrorMessage
+                BibEntry errorBibEntry = new BibEntry();
+                errorBibEntry.setField("html_representation", "An error occured. We are sorry for that.");
+                BibDatabase errorBibDataBase = new BibDatabase();
+                errorBibDataBase.insertEntry(errorBibEntry);
+                parserResult = new ParserResult(errorBibDataBase);
+            }
+        } catch (IOException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        return parserResult.getDatabase().getEntries();
+    }
+
+    /**
+     * Contact the server with the title of the selected item
+     * @param query
+     * @return
+     */
+    private String makeServerRequest(String query) {
+        query = constructQuery(query);
+        String response = "";
+
+        //Makes a request to the RESTful MDL-API. Example document.
+        //Servers-side functionality in implementation.
+
+        SSLContext sslContext = null;
+
+        try {
+            sslContext = SSLContexts.custom().loadTrustMaterial(null, new TrustSelfSignedStrategy()).build();
+        } catch (KeyManagementException | NoSuchAlgorithmException | KeyStoreException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+
+        SSLConnectionSocketFactory sslSocketFacktory = new SSLConnectionSocketFactory(sslContext);
+
+        try (CloseableHttpClient httpclient = HttpClients.custom().setSSLSocketFactory(sslSocketFacktory).build()) {
+            Unirest.setHttpClient(httpclient);
+            try {
+                response = Unirest.get(query).asString().getBody();
+            } catch (UnirestException e) {
+                LOGGER.error(e.getMessage(), e);
+                return "could not get a response.";
+            }
+
+            //Conversion. Server delivers false format, conversion here, TODO to fix
+            response = response.replaceAll("&gt;", ">");
+            response = response.replaceAll("&lt;", "<");
+        } catch (IOException e1) {
+            LOGGER.error(e1.getMessage(), e1);
+        }
+        return response;
+    }
+
+    /**
+     * Constructs the query based on title of the bibentry. Adds statistical stuff to the url.
+     * @param query: the title of the bib entry.
+     * @return
+     */
+    private String constructQuery(String query) {
+        StringBuffer queryBuffer = new StringBuffer();
+        //queryBuffer.append("https://api-dev.mr-dlib.org/v1/documents/");
```

Comment:
Before merging, all commented-out code (here and below) will have to be removed.